app/services/core_data_connector/import/instances.rb

```diff
@@ -0,0 +1,116 @@
+module CoreDataConnector
+  module Import
+    class Instances < Base
+      def cleanup
+        super
+
+        execute <<-SQL.squish
+          WITH update_instances AS (
+            UPDATE
+            core_data_connector_instances
+              SET z_instance_id = NULL
+          )
+
+          UPDATE
+          core_data_connector_names
+            SET z_source_id = NULL,
+                z_source_type = NULL
+        SQL
+      end
+
+      def load
+        super
+
+        execute <<-SQL.squish
+          UPDATE core_data_connector_instances instances
+             SET z_instance_id = z_instances.id,
+                 user_defined = z_instances.user_defined,
+                 updated_at = current_timestamp
+            FROM #{table_name} z_instances
+           WHERE z_instances.instance_id = instances.id
+        SQL
+
+        execute <<-SQL.squish
+        WITH
+
+        insert_instances AS (
+          INSERT INTO core_data_connector_instances (
+            project_model_id,
+            uuid,
+            z_instance_id,
+            user_defined,
+            created_at,
+            updated_at
+          )
+          SELECT z_instances.project_model_id, 
+                 z_instances.uuid, 
+                 z_instances.id, 
+                 z_instances.user_defined, 
+                 current_timestamp, 
+                 current_timestamp
+          FROM #{table_name} z_instances
+          WHERE z_instances.instance_id IS NULL
+          RETURNING id AS instance_id, z_instance_id
+
+        ),
+
+        insert_names AS (
+          INSERT INTO core_data_connector_names
+            ("name", z_source_id, z_source_type, created_at, updated_at)
```

Comment:
Like the other SQL, let's either keep this on one line or one column per line. Also, is there a need to put quotes around "name"?